lib/db.js

```diff
@@ -211,8 +220,24 @@ module.exports = (
   }
 
   DB.prototype.sessions = function (uid) {
+    const promises = [
+      this.pool.get('/account/' + uid + '/sessions')
+    ]
+
+    if (this.redis) {
+      promises.push(this.redis.getAsync(uid))
+    }
+
     log.trace({ op: 'DB.sessions', uid: uid })
-    return this.pool.get('/account/' + uid + '/sessions')
+    return P.all(promises)
+      .then(results => {
+        const tokens = results[0]
+        const redisTokens = results[1] ? JSON.parse(results[1]) : []
```

Comment:
Cannot fail since redis either returns null or what we have previously stored, and we explicitly store `json.stringified` zobjects